enhancements/siteconfig-extra-manifest-version-independence.md

```diff
@@ -0,0 +1,339 @@
+
+## Release Signoff Checklist
+
+- [ ] Enhancement is `implementable`
+- [ ] Design details are appropriately documented from clear requirements
+- [ ] Test plan is defined
+- [ ] Documentation is updated in the upstream
+- [ ] Documentation is updated in the downstream
+
+
+## Summary
+
+Siteconfig allows customer to include custom extra manifests and list them in the siteconfig CR to be added during CR generation. It provides the flexibility to add additional CRs from the customer point of view during install time along with the CRs the RAN team provides. 
+
+The enhancement seeks to address the version independence of extra manifests, by putting version dependent manifests in version specific folder in git and pointing the directory via **extraManifestPath** and disabling the fetching builtIn RAN CRs provided from the ztp container 
+
+
+## Motivation
+
+* **Currently** customer can put custom manifests in the git repo and point them via **extraManifestPath**. 
+* **Current behavior:** If customer only want to include custom CRs to build site specific CRs:
+  * **inclusionDefault** must be set to *exclude* which will exclude all the CRs from ztp-container
+  * pass the list of CR file names under **filter.include**, that will include listed CRs.
+  * but it does not allow to have the [same CR name](https://github.com/openshift-kni/cnf-features-deploy/blob/master/ztp/siteconfig-generator/siteConfig/siteConfigBuilder.go#L462-L464) for the custom CRs. 
+
+  * to be able to add custom CRs, customer needs to change the name of the custom CR.
+
+
+  ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      filter:
+        inclusionDefault: exclude
+        include:
+          - myCR.yaml					
+  ```
+
+  * as a result same named CR coming from user directory doesn't take precedent, rather siteconfig issues an error complaining same named CR is not allowed.
+
+### Goals
+
+- Allow customer to include any custom CRs having same/different name than builtIn CRs residing in the container
+- Provide a better control on how to enable and disable builtIn CRs
+- Keep backward compatibility of the behavior corresponds to 4.13 version.
+
+### Non-Goals
+
+- TBD
+### User Stories
+
+- [EPIC LINK](https://issues.redhat.com/browse/CNF-7365) --> [User Story](https://issues.redhat.com/browse/CNF-8672)
+
+## Proposal
+
+* **Controllability:** The proposal is to introduce a parameter called  <span style="color:red">*extraManifests.builtinCRs*</span>. which can hold value "*include|exclude*", which provide the flexibility for the customer to include/exclude CRs that is provided via ztp-container. 
+
+  * In this way, customer have full control of what CRs goes into the siteconfig. Until 4.13 release, customer couldn't put same named CR to the CR provided via ztp-container in the *extraManifestPath*. 
+
+
+ ```yaml
+  - cluster:
+    extraManifestPath: mypath/
+    extraManifests:
+      builtinCRs: include | exclude   <----- New field
```

Comment:
can we be more explicit instead of `builtinCRs`? `builtinCRs` implies CRs coming from currently attached site-gen conatiner's `~/ztp-container's ztp/source-crs/extra-manifest`? how about `siteGenContainerExtraManifest` ?

Reply:
This name seemed long!